ssl/quic/quic_channel_local.h

```diff
@@ -288,6 +292,8 @@ struct quic_channel_st {
 
     /* We have received transport parameters from the peer. */
     unsigned int                    got_remote_transport_params    : 1;
+    /* We have generated our local transport parameters. */
+    unsigned int                    got_local_transport_params     : 1;
```

Comment:
The quic_channel_st structure is 25 cachelines long.  It might be worth considering changing some of the implementation to save some space and accelerate load times of this structure.  Specifically:
1) changing the bitfields to to be a uint32_t and using bitmasks would save a 22 bit hole
2)  Moving the cur_local_cid to be immediately below the cur_remote_dcid would eliminate a 4 byte hole
3) moving the diag_num_rx_ack uint16_t to be below rx_ack_delay_exp would save 2 bytes

Its not alot, but at the size its at, it may goose performance a bit

Reply:
Ah, another fan of the lost art of C structure packing...

```
$ clang -Xclang -fdump-record-layouts -c ... ssl/quic/quic_channel.c
```
is useful here.

2 and 3 are valid, not sure I follow 1. Packing seems optimal here.

Reply:
Yeah, re 1, I don't quite understand how a series of bitfields of a total of 9 bits converted to a uint32_t would remove a 22 (or perhaps 23) bit hole.  Doesn't make sense to me.

Reply:
(aren't bitfields a language feature that essentially does the bit masking and shifting for you, so you don't have to do it manually?)

Reply:
Wait, how is this uint64_t? This is unsigned int, so 32 bits

Reply:
the type is unisigned int, but we have more than 32 bits of fields, for the compiler allocates 64 bits of storage